File: cs/HAC.Tests.Unit/ArrayMaximalAdjacentDifference.Test.cs
using NUnit.Framework;
using static HAC.HAC;

namespace HAC.Tests.Unit
{
    public class ArrayMaximalAdjacentDifferenceTest
    {
        [TestCase(new[] {2, 4, 1, 0}, 3)]
        [TestCase(new[] {2, 9, 1, 0}, 8)]
        public void ShouldWork(int[] data, int expected)
        {
            Assert.AreEqual(expected, ArrayMaximalAdjacentDifference(data));
        }
    }
}

File: php/tests/ArrayMaximalAdjacentDifferenceTest.php
<?php

use PHPUnit\Framework\TestCase;

require "src/array_maximal_adjacent_difference.php";

final class ArrayMaximalAdjacentDifferenceTest extends TestCase
{
    public function testArrayMaximalAdjacentDifference1()
    {
        $data = [2, 4, 1, 0];
        $actual = array_maximal_adjacent_difference($data);
        $expected = 3;
        $this->assertEquals($expected, $actual);
    }

    public function testArrayMaximalAdjacentDifference2()
    {
        $data = [2, 9, 1, 0];
        $actual = array_maximal_adjacent_difference($data);
        $expected = 8;
        $this->assertEquals($expected, $actual);
    }
}